packages/allure-cypress/src/browser/patching.ts

```diff
@@ -0,0 +1,130 @@
+import { isPromise } from "allure-js-commons/sdk";
+import type { CypressSuiteFunction, DirectHookImplementation, HookImplementation } from "../types.js";
+import {
+  completeSpecIfNoAfterHookLeft,
+  completeSpecOnAfterHookFailure,
+  throwAfterSpecCompletion,
+} from "./lifecycle.js";
+
+/**
+ * Patches the `after` function, to inject reporting of spec-level
+ * `after` hooks defined by the user.
+ */
+export const enableScopeLevelAfterHookReporting = () => {
+  const [getSuiteDepth, incSuiteDepth, decSuiteDepth] = createSuiteDepthCounterState();
+  patchDescribe(incSuiteDepth, decSuiteDepth);
+  patchAfter(getSuiteDepth);
+};
+
+const createSuiteDepthCounterState = (): [get: () => number, inc: () => void, dec: () => void] => {
+  let suiteDepth = 0;
+  return [
+    () => suiteDepth,
+    () => {
+      suiteDepth++;
+    },
+    () => {
+      suiteDepth--;
+    },
+  ];
+};
+
+const patchDescribe = (incSuiteDepth: () => void, decSuiteDepth: () => void) => {
```

Comment:
Here we can just pass the object I described above

Reply:
Done